### PR TITLE
Datadog wrapper

### PR DESCRIPTION
**Pull request by gherceg**
Enables the ability to add tags "as you go" to datadog, rather than having to fetch specific information at the end of the request resulting in slower (time) requests. 

This is inspired by the way Sentry's client works, in that you can add a tag to the current context or scope, so that all requests sent within that context contain this tag. In this case, the `FormplayerDatadog` is scoped for each request, so a fresh object should be used for each request, and any information that we would like to add to datadog can be tagged along the way, and sent at the end of the request.

### CODE REVIEW

File: src/main/java/org/commcare/formplayer/application/WebAppContext.java [hunk 1]
```diff
@@ -247,6 +248,13 @@ public FormplayerSentry raven() {
         return raven;
     }
 
+    @Bean(destroyMethod="cleanup")
```
Comment (calellowitz) [inline src/main/java/org/commcare/formplayer/application/WebAppContext.java:251]: What does this do?

**Review by calellowitz — COMMENTED** [review]
One question but looks good

File: src/main/java/org/commcare/formplayer/application/WebAppContext.java (hunk 1 above)
Comment (gherceg) [reply src/main/java/org/commcare/formplayer/application/WebAppContext.java]: Ah definitely don't need this. Didn't refine my copy/paste for the `FormplayerSentry` object, but looks like it just specifies to call that method (`cleanup`) when destroying the object. The Datadog Client file doesn't have any special cleanup, so I don't think the `FormplayerDatadog` class needs any either.

**Comment by gherceg** [comment]
@calellowitz this is ready to be reviewed. Verified that request metrics are still coming in on staging after applying this change.

**Review by calellowitz — APPROVED** [review]

**Review by snopoke — COMMENTED** [review]
I don't see anywhere in this PR we it's being used in the way you describe in the PR description. Are there more changes coming later?

**Comment by snopoke** [comment]
> I don't see anywhere in this PR we it's being used in the way you describe in the PR description. Are there more changes coming later?

https://github.com/dimagi/formplayer/pull/851

**Comment by gherceg** [comment]
@snopoke updated the description to be more exact.

**Comment by snopoke** [comment]
@gherceg have you looked Spring's built in metrics support? https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-features.html#production-ready-metrics

**Comment by gherceg** [comment]
I haven't looked into built in metrics support. We use this for Spring specific metrics already right? Are you specifically interested in using Spring's [Micrometer](https://micrometer.io/) to decouple us from datadog?

**Comment by snopoke** [comment]
@gherceg I wasn't thinking specifically of decoupling us from DD, just avoiding reinventing the wheel. I haven't looked at Micrometer in depth so I don't know if it solves any of these issues at all or makes them easier.

**Pull request merged**